src/components/TableComponent/index.jsx

```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { Pagination, Table } from '@edx/paragon';
+import 'font-awesome/css/font-awesome.css';
+
+import LoadingMessage from '../LoadingMessage';
+import TableLoadingOverlay from '../TableLoadingOverlay';
+import StatusAlert from '../StatusAlert';
+import './TableComponent.scss';
+
+class TableComponent extends React.Component {
+  componentDidMount() {
+    // Get initial data
+    this.props.paginateTable();
+  }
+
+  renderTableContent() {
+    const {
+      className,
+      currentPage,
+      pageCount,
+      tableSortable,
+      data,
+      ordering,
+      formatData,
+      paginateTable,
+      sortTable,
+      id,
+      loading,
+    } = this.props;
+
+    const columnConfig = this.props.columns.map(column => ({
+      ...column,
+      onSort: column.columnSortable ? (direction) => {
+        sortTable({ key: column.key, direction });
+      } : null,
+    }));
+
+    let sortDirection;
+    let sortColumn;
+    if (tableSortable) {
+      sortDirection = ordering && ordering.indexOf('-') !== -1 ? 'desc' : 'asc';
+      sortColumn = (ordering && ordering.replace('-', '')) || columnConfig[0].key;
+    }
+
+    return (
+      <div className={className}>
+        <div className="row">
+          <div className="col">
+            { loading && <TableLoadingOverlay /> }
+            <div className="table-responsive">
+              <Table
+                id={id}
+                className={['table-sm', 'table-striped']}
+                columns={columnConfig}
+                data={formatData(data)}
+                tableSortable={tableSortable}
+                defaultSortedColumn={sortColumn}
+                defaultSortDirection={sortDirection}
+              />
+            </div>
+          </div>
+        </div>
+        <div className="row mt-2">
+          <div className="col d-flex justify-content-center">
+            <Pagination
+              paginationLabel={`${id}-pagination`}
+              pageCount={pageCount}
+              currentPage={currentPage}
+              onPageSelect={paginateTable}
+            />
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  renderLoadingMessage() {
+    return <LoadingMessage className="table-loading" />;
+  }
+
+  renderErrorMessage() {
+    return (
+      <StatusAlert
+        alertType="danger"
+        iconClassName={['fa', 'fa-times-circle']}
+        title="Unable to load full report"
```

Comment:
Similar to the comment above -- do we want to be able to customize error messages/titles in `TableComponent`? Right now, every table using this component would have the title "Unable to load full report."

Reply:
Will make this more generic. As above, we can follow up with custom messages as props if we decide to